boards/shared/Src/App/App_SharedStateMachine.c

```diff
@@ -10,6 +11,38 @@ struct StateMachine
     struct World *      world;
 };
 
+/**
+ * Run the given tick function over the given state machine if the tick function
+ * is not null
+ *
+ * @param state_machine The state machine to run the tick function over
+ * @param tick_function The tick function to run over the state machine
+ */
+void App_SharedStateMachine_RunStateTickFunctionIfNotNull(
+    struct StateMachine *const state_machine,
+    void (*tick_function)(struct StateMachine *))
+{
+    if (tick_function == NULL)
+    {
+        // Nothing to do
+        return;
+    }
+
+    tick_function(state_machine);
+
+    // Check if we should transition states
+    if (state_machine->next_state != state_machine->current_state)
```

Comment:
Can we have write a quick test to make sure that when `Tick1Hz` changes the state from `State A` to `State B`, the next `Tick1kHz` will tick in `State B` as well? In other words, the state change is propagated between the different tick functions.

Reply:
Adding `Test_SharedStateMachine` under `shared/Test/Src` would be great

Reply:
Agreed. Added.